### ISSUE TITLE
Ray-wise High-Confidence Noise Cannot Be Filtered by Masks During Fine-Tuning on Custom Dataset

### ISSUE THREAD

**Issue by lnSong** [facebookresearch/map-anything#104]
Thank you for your excellent work!
When fine-tuning the model on my custom dataset, I encountered an issue with ray-wise noise (noise along the ray direction). These noises have high confidence scores, making them impossible to filter out using masks. Have you encountered such a problem before? Is this a training-related issue or a dataset-related problem? Thank you very much for your help!

<img width="2001" height="1375" alt="Image" src="https://github.com/user-attachments/assets/a01b4d05-3a75-4518-8107-132e75359026" />

<img width="2001" height="1375" alt="Image" src="https://github.com/user-attachments/assets/ae72d253-b554-404c-a1fd-5f3b3d35693f" />

**Comment by lnSong** [comment]
Could this be an underfitting issue? I set the number of epochs to 100 and the batch size per GPU to 2. When the dataset contains 100 samples, there is very little noise; however, when the dataset size increases to 1000 samples, the noise becomes significantly more prominent.

**Comment by lnSong** [comment]
In the above results, I used only the non_ambiguous_mask. When I also employ the edge_mask, it provides a certain level of filtering against these noises.

**Comment by lnSong** [comment]
However, when using the original .pth file before fine-tuning, it didn't produce this much noise—the results were only somewhat inaccurate in terms of scale/distance. After fine-tuning, the noise actually increased. Why did this happen?

**Comment by Nik-V9** [comment]
Hi, I would recommend mixing in synthetic data from the MapA training suite when training on your own data. This helps with robust optimization despite the presence of noisy GT.

**Comment by lnSong** [comment]
Thank you for your kind help!
